frontend/controller/actions/mailbox.js

```diff
@@ -31,18 +31,26 @@ export default (sbp('sbp/selectors/register', {
     try {
       const rootState = sbp('state/vuex/state')
       const rootGetters = sbp('state/vuex/getters')
-      const partnerProfile = rootGetters.ourContactProfiles[params.data.username]
+      const partnerProfiles = params.data.usernames.map(username => rootGetters.ourContactProfiles[username])
+      const isPrivacyLevelPrivate = params.data.privacyLevel === CHATROOM_PRIVACY_LEVEL.PRIVATE
 
-      if (!partnerProfile) {
+      const hasEmptyProfile = partnerProfiles.some(profile => !profile)
+      if (hasEmptyProfile) {
         throw new GIErrorUIRuntimeError(L('Incorrect username to create direct message.'))
+      } else if (!partnerProfiles.length) {
+        throw new GIErrorUIRuntimeError(L('Not enough usernames provided.'))
+      } else if (isPrivacyLevelPrivate && partnerProfiles.length !== 1) {
+        throw new GIErrorUIRuntimeError(L('Incorrect number of usernames provided.')) // TODO: need to add translation
+      } else if (isPrivacyLevelPrivate && rootGetters.ourPrivateDirectMessages[params.data.usernames[0]]) {
+        throw new GIErrorUIRuntimeError(L('Already existing direct message channel.'))
```

Comment:
Is this an error that can be triggered by the user? If so, why? Couldn't we just immediately take them to that chatroom instead of giving an error? And if it's meant to be an error that the programmer sees, then it doesn't need to be translated.